fix: core not importable in client

ChristopherTrimboli opened the pull request:
Fixes client error:

```
Uncaught TypeError: Failed to resolve module specifier "@elizaos/core". Relative references must start with either "/", "./", or "../".
```

Also adds vite-node-polyfills to supplement Buffer and process missing.

tcm390 commented:
I'm getting the following error: define_Buffer_default.allocUnsafe is not a function. It might be related

<img width="630" alt="Screenshot 2025-05-15 at 10 48 54 AM" src="https://github.com/user-attachments/assets/ca78fcda-221b-4aff-b3eb-d47c39dc53aa" />

ChristopherTrimboli commented:
> I'm getting the following error: define_Buffer_default.allocUnsafe is not a function. It might be related. Looks like we may need to externalize it.
> 
> <img alt="Screenshot 2025-05-15 at 10 48 54 AM" width="630" src="https://private-user-images.githubusercontent.com/60634884/443920959-ca78fcda-221b-4aff-b3eb-d47c39dc53aa.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDcyNzgwODcsIm5iZiI6MTc0NzI3Nzc4NywicGF0aCI6Ii82MDYzNDg4NC80NDM5MjA5NTktY2E3OGZjZGEtMjIxYi00YWZmLWIzZWItZDQ3YzM5ZGM1M2FhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA1MTUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNTE1VDAyNTYyN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTdiZTc5MmFlMTM0MTJjYzM5YjVmM2NjNTUzNzk3MTg3M2Y3YmIwMTlhZWU1ZTMyMDAyODFjZDg3MDE2NjFkZTAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.MYoQfkXQNCSXb6C2v7Gfp8PJDGM_qiYRwhcQ6oAtErU">

fixed, added vite-node-polyfills

tcm390 commented:
<img width="1495" alt="Screenshot 2025-05-15 at 11 24 03 AM" src="https://github.com/user-attachments/assets/81489a8f-0fad-4c03-8aae-bea871567ac3" />


Hmmmm...got this

ChristopherTrimboli commented:
> <img alt="Screenshot 2025-05-15 at 11 24 03 AM" width="1495" src="https://private-user-images.githubusercontent.com/60634884/443929770-81489a8f-0fad-4c03-8aae-bea871567ac3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDcyODA0MDcsIm5iZiI6MTc0NzI4MDEwNywicGF0aCI6Ii82MDYzNDg4NC80NDM5Mjk3NzAtODE0ODlhOGYtMGZhZC00YzAzLThhYWUtYmVhODcxNTY3YWMzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA1MTUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNTE1VDAzMzUwN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTczMDM4M2RjOGRmZmZlMmE1OTNmODhkZDBkMTUwYzcyNjU2ZWU3MTgwNDc5NDY2ODNhNmYzNzI1MDlmYmNjZTQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.SJiJUqcGPBSMEHcYpbIJV05iXH2cFTKcaF6-VCOCaHA">
> Hmmmm...got this

some race with copy dist, polyfills taking longer to build index.html, I added more specific check to `copyClientDist`, seems like works now.

tcm390 reviewed (approved):
I tested it and it’s working well on my end!
